src/packages/dgswap/metadata.ts

```diff
@@ -1,50 +1,376 @@
-import {z} from 'zod';
+import { z } from 'zod';
 
 /* Function names are same as the function names in the services.ts file */
-export const getPoolByTokenSymbol = {
-    name: 'get_pool_by_token_symbol',
+// for params `where` schema, please refer to https://thegraph.com/explorer/subgraphs/DFu3UKnkVWq4xgYq5NFerMu6puA9SkqdMyjjWmauwqqM?view=Query&chain=arbitrum-one
+// pools
+export const getPools = {
+    name: 'get_pools',
     description: 'Get the dragonswap pool by token symbol',
     params: z.object({
-        symbol0: z.string(),
-        symbol1: z.string()
+        count: z.number(),
+        skip: z.number(),
+        where: z.object({}).optional()
```

Comment:
for where params, user can use the same schema as they use in subgraph playground. we don't need the validation here since the graphql endpoint will return any validation errors related to where field.

Reply:
HI @Sotatek-TanHoang , these are called from ai frameworks like vercel, langchain etc. i would recommend to use simple data base types instead of like a query. Last time yeah the union didnot work as well .

Reply:
Thanks for your suggestion.I agree on using simple database types make developer experience better but the amount of work to rewrite the params logic is quite a lot.
If the current issue with a single where params is the lack of inner types, we can use graphql schema from dgswap to generated typescript types. What is your opinion about this? @praveen-kaia